galite-core/src/main/kotlin/org/kopi/galite/report/VReport.kt

```diff
@@ -15,6 +15,612 @@
+   */
+  protected fun hasTrigger(event: Int, index: Int = 0): Boolean = VKT_Triggers!![index][event] != 0
+
+  fun setMenu() {
+    if (!built) {
+      // only when commands are displayed
+      return
+    }
+    val column = getSelectedColumn()
+    val (x, y) = getSelectedCell()
+    val foldEnabled = column != -1 && !model.isColumnFold(column) || x != -1 && y != -1 && !model.isRowFold(y, x)
+    val unfoldEnabled = column != -1 || x != -1 && y != -1
+
+    if (cmdFold != null) {
+      setCommandEnabled(cmdFold!!, foldEnabled)
+    }
+    if (cmdUnfold != null) {
+      setCommandEnabled(cmdUnfold!!, unfoldEnabled)
+    }
+    if (cmdSort != null) {
+      setCommandEnabled(cmdSort!!, column != -1)
+    }
+    if (cmdOpenLine != null) {
+      setCommandEnabled(cmdOpenLine!!, model.isRowLine(y))
+    }
+    if (cmdFoldColumn != null) {
+      setCommandEnabled(cmdFoldColumn!!, column != -1)
+    }
+    if (cmdUnfoldColumn != null) {
+      setCommandEnabled(cmdUnfoldColumn!!, column != -1)
+    }
+    if (cmdColumnInfo != null) {
+      setCommandEnabled(cmdColumnInfo!!, column != -1)
+    }
+    if (cmdEditColumn != null) {
+      setCommandEnabled(cmdEditColumn!!, column != -1 && model.getAccessibleColumn(column)!!.addedAtRuntime)
+    }
+  }
+  // ----------------------------------------------------------------------
+  // IMPLEMENTATION
+  // ----------------------------------------------------------------------
+  /**
+   * Returns the selected column or -1 if no column is selected.
+   */
+  private fun getSelectedColumn(): Int = (getDisplay() as UReport).getSelectedColumn()
+
+  /**
+   * Returns the selected cell or !!! ??? if no cell is selected.
+   */
+  private fun getSelectedCell(): Point = (getDisplay() as UReport).getSelectedCell()
+
+  fun genHelp(): String? {
+    val surl = StringBuffer()
+    val fileName: String? = VHelpGenerator().helpOnReport(pageTitle,
+            commands,
+            model,
+            help)
+
+    return if (fileName == null) {
+      null
+    } else {
+      try {
+        surl.append(File(fileName).toURL().toString())
+      } catch (mue: MalformedURLException) {
+        throw InconsistencyException(mue)
+      }
+      surl.toString()
+    }
+  }
+
+  fun showHelp() {
+    VHelpViewer().showHelp(genHelp())
+  }
+
+  // ----------------------------------------------------------------------
+  // DATA MEMBERS
+  // ----------------------------------------------------------------------
+  private var cmdFold: VCommand? = null
+  private var cmdUnfold: VCommand? = null
+  private var cmdSort: VCommand? = null
+  private var cmdOpenLine: VCommand? = null
+  private var cmdFoldColumn: VCommand? = null
+  private var cmdUnfoldColumn: VCommand? = null
+  private var cmdColumnInfo: VCommand? = null
+  private var cmdEditColumn: VCommand? = null
+
+  /**
+   * Set the source for this document
+   */
+  private lateinit var source: String
+  override fun getSource(): String? = null
+
```

Comment:
I dont agree

Reply:
Changed to : 
Override fun getSource() : String = this.source